### PR TITLE
Create Rule “optimise-js-with-lodash/rule”

### PR DESCRIPTION
**Pull request by JackDevAU**
Automatically generated by Netlify CMS

### CODE REVIEW

File: rules/optimise-js-with-lodash/rule.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+type: rule
+title: Do You Optimise Your JavaScript Code with Lodash Wisely and Efficiently?
```
Comment (bradystroud) [inline rules/optimise-js-with-lodash/rule.md:3]: Use sentence case - keep it consistent to other rules

File: rules/optimise-js-with-lodash/rule.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+Here's an example where using Lodash's `_.map` method is unnecessary:
+
+::: bad
+```js
+const arr = [1, 2, 3];
+const newArr = _.map(arr, function(n) { return n * 3; });
+console.log(newArr);
+// output: [3, 6, 9]
+```
+**Figure: using Lodash's _.map method**
```
Comment (bradystroud) [inline rules/optimise-js-with-lodash/rule.md:41]: See https://www.ssw.com.au/rules/rule/#captions-on-boxes for the best approach

**Review by bradystroud — CHANGES_REQUESTED** [review]

File: rules/optimise-js-with-lodash/rule.md (hunk 1 above)
Comment (JackDevAU) [inline rules/optimise-js-with-lodash/rule.md:3]:
```suggestion
title: Do you optimise your JavaScript code with Lodash wisely and efficiently?
```

File: rules/optimise-js-with-lodash/rule.md (hunk 2 above)
Comment (JackDevAU) [reply rules/optimise-js-with-lodash/rule.md]: Done!

File: rules/optimise-js-with-lodash/rule.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+---
+Do you want to make your JavaScript code more efficient and easier to read? Lodash might be the utility library you need. But like any tool, knowing when and how to use it is crucial for the maintainability and performance of your project.
+
+JavaScript is a powerful language, but it can sometimes be tricky to write clean, efficient code. This is where Lodash shines. Lodash is a JavaScript utility library providing handy methods for manipulating and combining arrays, objects, numbers, strings, among other data types.
+
+<!--endintro-->
+
+::: info  
+For optimal bundle size and tree-shaking capabilities in modern JavaScript applications, we recommend using [lodash-es](https://www.npmjs.com/package/lodash-es) over the standard lodash library.
+:::
```
Comment (bradystroud) [inline rules/optimise-js-with-lodash/rule.md:19]: cc @christoment 

can you do tree shaking with standard one?
This one looks a bit outdated maybe

**Review by bradystroud — CHANGES_REQUESTED** [review]
cc @christoment 
lodash is massive, you should cherry pick what functions you are importing.
Can you update the rule to add this performance tip?

File: rules/optimise-js-with-lodash/rule.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  { 'user': 'fred',    'age': 40, 'active': false },
+  { 'user': 'pebbles', 'age': 1,  'active': true }
+];
+
+const user = _.find(users, { 'age': 1, 'active': true });
+console.log(user);
+// output: { 'user': 'pebbles', 'age': 1, 'active': true }
+
+```
+Figure: Lodash's _.find method makes this a breeze:
```
Comment (JackDevAU) [inline rules/optimise-js-with-lodash/rule.md:83]:
```suggestion
Figure: Lodash's _.find method 
```

File: rules/optimise-js-with-lodash/rule.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  { 'user': 'barney',  'age': 36, 'active': true },
+  { 'user': 'fred',    'age': 40, 'active': false },
+  { 'user': 'pebbles', 'age': 1,  'active': true }
+];
+
+const user = _.find(users, { 'age': 1, 'active': true });
+console.log(user);
+// output: { 'user': 'pebbles', 'age': 1, 'active': true }
+
+```
```
Comment (JackDevAU) [inline rules/optimise-js-with-lodash/rule.md:82]:
```suggestion
```js
import { find } from 'lodash-es';

const users = [
  { 'user': 'barney',  'age': 36, 'active': true },
  { 'user': 'fred',    'age': 40, 'active': false },
  { 'user': 'pebbles', 'age': 1,  'active': true }
];

const user = find(users, { 'age': 1, 'active': true });
console.log(user);
// output: { 'user': 'pebbles', 'age': 1, 'active': true }

```
```

Comment (JackDevAU) [reply rules/optimise-js-with-lodash/rule.md:82]: adds good example of import using lodash-es

**Review by bradystroud — APPROVED** [review]

**Pull request merged**